src/arkode/arkode_mristep.c

```diff
@@ -1082,42 +1078,65 @@ int mriStep_Init(void* arkode_mem, int init_type)
                       "Error in coupling table");
       return (ARK_ILL_INPUT);
     }
-
+    
     /* stage types */
-    if (step_mem->stagetypes)
+    step_mem->stagetypes = realloc(step_mem->stagetypes,
+                                   step_mem->stages * sizeof(*step_mem->stagetypes));
+    if (step_mem->stagetypes == NULL)
     {
-      free(step_mem->stagetypes);
-      step_mem->stagetypes = NULL;
-      ark_mem->liw -= step_mem->stages;
+      arkProcessError(ark_mem, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
+                      MSG_ARK_MEM_FAIL);
+      return (ARK_MEM_FAIL);
     }
-    step_mem->stagetypes = (int*)calloc(step_mem->stages, sizeof(int));
-    ark_mem->liw += step_mem->stages;
+    ark_mem->liw += delta_stages;
     for (j = 0; j < step_mem->stages; j++)
     {
       step_mem->stagetypes[j] = mriStepCoupling_GetStageType(step_mem->MRIC, j);
     }
-
+    
     /* explicit RK coefficient row */
-    if (step_mem->Ae_row)
+    step_mem->Ae_row = realloc(step_mem->Ae_row,
+                               step_mem->stages * sizeof(*step_mem->Ae_row));
```

Comment:
I'm okay with changing to `realloc`, but (a) please cast the result to the correct type (`sunrealtype*`), and (b) `memset` the new memory to all zeros to retain `calloc`'s initialization feature.